components/automate-ui/src/app/modules/user/user-details/user-details.component.ts

```diff
@@ -34,99 +41,183 @@ export type UserTabName = 'password' | 'details';
+        ]).subscribe(([routeId, currentUserId]: [string, string]) => {
+          if (routeId === currentUserId) {
+            this.userDetails = new UserAdminSelfDetails(
+              this.store, this.layoutFacade, this.isDestroyed, this.fb);
+          } else {
+            this.userDetails = new UserAdminDetails(routeId,
+              this.store, this.layoutFacade, this.isDestroyed, this.fb);
+          }
+          this.loading = false;
+        });
+      }
+    });
 
-    this.store.select(routeURL).pipe(takeUntil(this.isDestroyed))
-      .subscribe((url: string) => {
-        this.url = url;
-        const [, fragment] = url.split('#');
-        this.tabValue = (fragment === 'password') ? 'password' : 'details';
-      });
+    this.store.select(routeURL).pipe(
+      takeUntil(this.isDestroyed)
+    ).subscribe((url: string) => {
+      this.url = url;
+      const [, fragment] = url.split('#');
+      this.tabValue = (fragment === 'password') ? 'password' : 'details';
+    });
   }
 
   ngOnDestroy(): void {
     this.isDestroyed.next(true);
     this.isDestroyed.complete();
   }
 
-  private createForms(fb: FormBuilder): void {
+  public onSelectedTab(event: { target: { value: UserTabName } }): void {
+    this.tabValue = event.target.value;
+    // Drop the previous fragment and add the incoming fragment.
+    this.router.navigate([this.url.split('#')[0]], { fragment: event.target.value });
+  }
+
+  public handlePasswordInput(): void {
+    this.userDetails.passwordForm.get('confirmPassword').updateValueAndValidity();
+  }
+}
+
+abstract class UserDetails {
+  abstract loading$: Observable<boolean>;
+  abstract saveSuccessful: boolean;
+  abstract saveInProgress: boolean;
+  abstract user: User;
+  abstract displayNameForm: FormGroup;
+  abstract passwordForm: FormGroup;
+  abstract showBreadcrumbs: boolean;
+  abstract showPreviousPassword: boolean;
+
+  constructor() {}
+
+  abstract savePassword(): void;
+  abstract saveDisplayName(): void;
+  protected abstract createPasswordForm(fb: FormBuilder): void;
+
+  protected resetForms(): void {
+    this.displayNameForm.reset();
+    if (this.user) {
+      this.displayNameForm.patchValue({displayName: this.user.name});
+    }
+    this.passwordForm.reset();
+  }
+
+  protected createForms(fb: FormBuilder): void {
     // Must stay in sync with error checks in user-details.component.html
     this.displayNameForm = fb.group({
       displayName: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
     });
+
+    this.createPasswordForm(fb);
+  }
+}
+
+// This view is used when a user goes to another users detail page.
+class UserAdminDetails extends UserDetails {
+  public loading$: Observable<boolean>;
+  public user: User;
+  public saveSuccessful = false;
+  public saveInProgress = false;
+  public displayNameForm: FormGroup;
+  public passwordForm: FormGroup;
+  public showBreadcrumbs = true;
+  public showPreviousPassword = false;
```

Comment:
Yes--looks better with an abstract class! 👍 
One more streamlining note here; while you can declare these as abstract properties and then instantiate them, there is no need because you are instantiating them identically. So you can move this block into the abstract class and not have to repeat it in each derived class.

Reply:
Thanks. I moved all the fields to the abstract class but the last two because they are different for each derived class.